neptunecontrib/monitoring/skopt.py

```diff
@@ -81,246 +95,95 @@ def log_results(results, experiment=None):
-
-
-def log_plot_evaluations(results, experiment=None, channel_name='evaluations'):
-    """Logs skopt plot_evaluations figure to neptune.
 
-    Image channel `evaluations` is created and the output of the
-    plot_evaluations function is first covented to `neptune.Image` and
-    then sent to neptune.
-
-    Args:
-        results('scipy.optimize.OptimizeResult'): Results object that is typically an
-            output of the function like `skopt.forest_minimize(...)`
-        experiment(`neptune.experiments.Experiment`): Neptune experiment. Default is None.
-
-    Examples:
-        Run skopt training::
-
-            ...
-            results = skopt.forest_minimize(objective, space,
-                                base_estimator='ET', n_calls=100, n_random_starts=10)
+def _log_plot_regret(results, experiment=None):
+    fig, ax = plt.subplots()
+    sk_plots.plot_regret(results, ax=ax)
+    experiment.log_image('plot_regret', fig)
 
-        Send skopt plot_evaluations figure to neptune::
-
-            import neptune
-            import neptunecontrib.monitoring.skopt as sk_utils
-
-            neptune.init(project_qualified_name='USER_NAME/PROJECT_NAME')
-
-            sk_utils.log_plot_evaluations(results)
-
-    """
-    _exp = experiment if experiment else neptune
 
+def _log_plot_evaluations(results, experiment=None):
     fig = plt.figure(figsize=(16, 12))
     fig = axes2fig(sk_plots.plot_evaluations(results, bins=10), fig=fig)
+    experiment.log_image('plot_evaluations', fig)
 
-    with tempfile.NamedTemporaryFile(suffix='.png') as f:
-        fig.savefig(f.name)
-        _exp.send_image(channel_name, f.name)
-
-
-def log_plot_objective(results, experiment=None, channel_name='objective'):
-    """Logs skopt plot_objective figure to neptune.
-
-    Image channel `objective` is created and the output of the
-    plot_objective function is first covented to `neptune.Image` and
-    then sent to neptune.
-
-    Args:
-        results('scipy.optimize.OptimizeResult'): Results object that is typically an
-            output of the function like `skopt.forest_minimize(...)`
-        experiment(`neptune.experiments.Experiment`): Neptune experiment. Default is None.
-
-    Examples:
-        Run skopt training::
 
-            ...
-            results = skopt.forest_minimize(objective, space,
-                                            base_estimator='ET', n_calls=100, n_random_starts=10)
+def _log_plot_objective(results, experiment=None):
+    try:
+        fig = plt.figure(figsize=(16, 12))
+        fig = axes2fig(sk_plots.plot_objective(results), fig=fig)
+        experiment.log_image('plot_objective', fig)
+    except Exception as e:
+        print('Could not create the objective chart due to error: {}'.format(e))
 
-        Send skopt plot_objective figure to neptune::
 
-            import neptune
-            import neptunecontrib.monitoring.skopt as sk_utils
+def _log_results_object(results, experiment=None):
```

Comment:
I'd feel a lot safer if `experiment` didn't default to `None`, since it's called in the next line :)

Reply:
Actually, this goes for all the helpers in this file.

Reply:
Yep, dropped when refactoring

Reply:
Done